modules/@apostrophecms/migration/index.js

```diff
@@ -38,15 +39,11 @@ module.exports = {
               manager.addSortifyMigration(field.name);
             });
           });
-        },
-        async executeMigrations() {
-          if ((process.env.NODE_ENV !== 'production') || self.apos.isNew) {
-            // Run migrations at dev startup (low friction).
-            // Also always run migrations at first startup, so even
-            // in prod with a brand new database the after event always fires
-            // and we get a chance to mark the migrations as skipped
-            await self.migrate(self.apos.argv);
-          }
+        }
+      },
+      before: {
```

Comment:
Invoke the new capability, which is not a registered, named migration, but rather an independent part of the migration phase that does its own checks to determine if work is needed.